### PR TITLE
Make user-agent string a configuration option

### PR DESCRIPTION
**Pull request by abraunegg**
* Make user-agent string a configuration option

### CODE REVIEW

**Comment by abraunegg** [comment]
@norbusan 
Right now the 'default' is something like this:
```
Phobos-std.net.curl/2.085 (libcurl/7.29.0)
```

What it, for 2.3.10, we make the default:
```
OneDrive Client for Linux v{version_number}
```

This would make more sense rather than the phobs / curl string at least.

We could then have a wiki page that links to browser 'user-agents' should that want to be changed. 

Thoughts?

File: src/onedrive.d [hunk 1]
```diff
@@ -98,7 +98,13 @@ final class OneDriveApi
 		if (cfg.getValueBool("debug_https")) {
 			http.verbose = true;
 			.debugResponse = true;
-        }
+		}
+
+		// Custom User Agent
+		if (cfg.getValueString("user_agent") != "") {
+			http.setUserAgent = cfg.getValueString("user_agent");
+		}
```
Comment (abraunegg) [inline src/onedrive.d]:
```
// Custom User Agent
		if (cfg.getValueString("user_agent") != "") {
			http.setUserAgent = cfg.getValueString("user_agent");
		} else {
			http.setUserAgent = "OneDrive Client for Linux " ~ strip(import("version"));
		}
```

Comment (abraunegg) [reply src/onedrive.d]: an option or no ?

Comment (norbusan) [reply src/onedrive.d]: Yes, strongly agree to do this! It also shows a bit more presence to the servers ;-)

**Pull request merged**